jobs/anac-certified-attributes-importer/src/service/processor.ts

```diff
@@ -0,0 +1,218 @@
+import { ANAC_ABILITATO_CODE, ANAC_INCARICATO_CODE, ANAC_IN_CONVALIDA_CODE } from '../config/index.js'
+import { parse } from 'csv/sync'
+import {
+  CsvRow,
+  NonPaRow,
+  PaRow,
+  PersistentTenant,
+  InteropContext,
+  AnacAttributes,
+  BatchParseResult,
+  AttributeIdentifiers,
+} from '../model/index.js'
+import { ReadModelQueries, SftpClient, TenantProcessService } from '../service/index.js'
+import { RefreshableInteropToken, zipBy, logError, logWarn } from '@interop-be-reports/commons'
+import crypto from 'crypto'
+
+export async function importAttributes(
+  sftpClient: SftpClient,
+  readModel: ReadModelQueries,
+  tenantProcess: TenantProcessService,
+  refreshableToken: RefreshableInteropToken,
+  recordsBatchSize: number,
+  anacTenantId: string
+): Promise<void> {
+  const jobCorrelationId = crypto.randomUUID()
+  const batchSize = recordsBatchSize
+
+  const fileContent = await sftpClient.downloadCSV()
+
+  const attributes: AnacAttributes = await getAttributesIdentifiers(readModel, anacTenantId)
+
+  const preparedProcessTenants = processTenants(tenantProcess, refreshableToken, attributes, jobCorrelationId)
+
+  let scanComplete = false
+  let fromLine = 1
+
+  do {
+    const batchResult: BatchParseResult = getBatch(fileContent, fromLine, batchSize, jobCorrelationId)
+
+    // TODO Not sure this is the best way to satisfy type checks
+    const paOrgs: PaRow[] = batchResult.records
+      .map((org: CsvRow) => {
+        if ('codice_ipa' in org && org.codice_ipa) return org as PaRow
```

Comment:
```suggestion
        if ('codice_ipa' in org) return org
```
`as PaRow` is not needed here. The typescript engine automatically infers the correct type from the union watching at the if statement `'codice_ipa' in org`.

I always tend to avoid the use of `as` since it shuts off the typecheck completely. 
Using `satisfies` instead of `as` is always a safer option. 

`as` tells typescript to blindly trust you about the type, no typecheck performed.
`satisfies` instead tells typescript to cast the type and also perform a typecheck.

Reply:
nice, I didn't know about `satisfies`, thanks!